src/job.rs

```diff
@@ -289,25 +291,42 @@ impl Job {
                             // should definitely go here, because user has to open the link to
                             // resume message sending.
-                            let msg_id = MsgId::new(self.foreign_id);
-                            message::set_msg_failed(context, msg_id, Some(err.to_string())).await;
-                            match Message::load_from_db(context, msg_id).await {
-                                Ok(message) => {
-                                    chat::add_info_msg(context, message.chat_id, err.to_string())
-                                        .await
-                                }
-                                Err(e) => error!(
-                                    context,
-                                    "couldn't load chat_id to inform user about SMTP error: {}", e
-                                ),
-                            };
                             Status::Finished(Err(format_err!("Permanent SMTP error: {}", err)))
                         }
                     }
-                    async_smtp::smtp::error::Error::Transient(_) => {
+                    async_smtp::smtp::error::Error::Transient(ref response) => {
                         // We got a transient 4xx response from SMTP server.
                         // Give some time until the server-side error maybe goes away.
-                        Status::RetryLater
+
+                        // Sometimes we receive the 4.1.2 error, which stays for "Bad destination system address"
+                        // or more simple, the domain doesn't exist or is invalid. This should be a permanent error
+                        // and fail more early.
+
+                        println!("xxx6 {:?}", response.message);
+                        if let Some(index) = response.message[0].find(" ") {
+                            let (extended_code, message) = response.message[0].split_at(index);
+
+                            let address = if let (Some(index_start), Some(index_end)) =
+                                (message.find("<"), message.find(">"))
+                            {
+                                Some(&message.get(index_start + 1..index_end))
+                            } else {
+                                None
+                            };
+
+                            let extended_error_bad_destination = "4.1.2";
+                            #[allow(clippy::absurd_extreme_comparisons)]
+                            if extended_code == extended_error_bad_destination && self.tries >= JobRetries::SMTP_ERROR_TRANSIENT_BAD_DESTINATION_SYSTEM_ADDRESS {
+                                info!(context, "Smtp-job #{} is a transient \"bad destination\" error. Lets let it fail immediatly.", self.job_id);
+                                Status::Finished(Err(
+                                    format_err!("{}", context.stock_string_repl_str(StockMessage::FailedSendingTo, address.unwrap_or("Unknown")).await)
```

Comment:
This is "too clever", IMO. It's possible that the message does not contain address in this format, only contains the domain, contains admin address in the form `if the problem persists, please report to <admin@example.org>` etc. Better just put the original error here, even though it is usually in English only.

We already do it for `Permanent SMTP error:` above.

Reply:
I don't know how standardized/similiar the error messages are accross different email servers. If we can figure out the email that is making trouble, this would be quite nice and helpful for the user. We could craft an error message a la "Can not send message to user@xxx. Probably this e-mail does not exist anymore or is temporary unavailable. Try to resend the message later or remove this user from this chat.". But of course only if it's standardized. I can remove it again, but I think better error messages would be helpful and important.

Reply:
The problem is that if heuristic fails, there is no access to the original error message except by looking into the logs. There is no guarantee at all that something between `<` and `>` is a destination email that caused the error or the email address at all.

With access to the original error user can at least search the web or DC forum for it.

Reply:
Ohh that makes sense. Simplifies my code. We can still think about this if it hits more people.